### DISCUSSION TITLE
Why is an empty string considered truthy?

### DISCUSSION THREAD

**Discussion by juniorforlife** [solidjs/solid#1717] [Q&A]
In [this](https://youtu.be/A5_58oenCqk?t=163) tutorial about `createResource` from SolidJS channel, the instructor said "but any other value including an empty string will trigger the resource".
Why is an empty string considered truthy in this case?

**Comment by snnsnn** [comment]
See the doc, guard clause is checking for null, false and undefined, not for falsy values:

> fetchData will be called as soon as sourceSignal has any value other than false, null, or undefined.
https://www.solidjs.com/docs/latest/api#createresource
